Automattic/jetpack
# Tiled Gallery: Accessibility - when images are not interactive, remove button role and tabindex.

Fixes https://github.com/Automattic/jetpack/issues/39925

## Proposed changes:

* This PR removes the button role and the tabindex that is added to Tiled Gallery images to help them be accessible (originally [added here](https://github.com/Automattic/jetpack/pull/37792)), but only in one scenario: when Carousel is disabled, and 'linkTo' is set to none. 
* One option was to remove the tabindex and role conditionally from the [relevant save.js file](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/extensions/blocks/tiled-gallery/gallery-image/save.js), but to prevent needing additional deprecation files as well as this only being a conditional change I opted to make this change when rendering the block via PHP. 

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/39925

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Replicating the issue:

1. On a test site (WoA / self-hosted), ensure the full-screen Carousel option is disabled under Jetpack > Settings > Writing: `/wp-admin/admin.php?page=jetpack#/writing`
2. Create a new post and add a Tiled Gallery block.
3. Ensure 'None' is set for the 'Link to' option in the sidebar:
<img src="https://github.com/user-attachments/assets/b01fe11b-0e6d-44d9-b942-a17b2deba651" width="250" />


4. Save and preview
5. If you inspect the image source for one of the Tiled Gallery images, you'll see a tabIndex of 0, and role of button:

![Image](https://github.com/user-attachments/assets/fd8c70b2-b3d2-4376-8b9b-1b2ceeb7ca28)

6. Click on any of the images - you'll see the image highlighted but no action taken
7. Similarly, tab onto one of the images and press enter. You'll see the image highlighted, but no action taken.
8. For all other options - images set to link to attachment type or media page, as well as Carousel enabled, you should be able to click on the image using a mouse and by tabbing and entering.
9. Changing between options and disabling / enabling Carousel should not result in any broken Tiled Gallery blocks.

Testing the fix

1. Repeat the above steps - step 8 should work as before, but for the testing steps where Carousel is disabled and the images set to link to 'none', you should not be able to tab onto the gallery images (nor see a role of button nor tabindex showing in the inspector for those specific gallery images).
2. Additionally, if you had a previously added Tiled Gallery, there should be no broken block messaging.

**Simple testing**

* Testing is not quite as easy on Simple - on trunk itself you may see a block validation error in the console upon adding a TG, saving, and refreshing the editor. However we can confirm the changes on the front-end, and that there is no visible broken TG in the editor itself with the changes (after applying the fix using the command in the generated comment below).
  *  In Simple, the `role="button"` is what should be removed on the front-end when Carousel is disabled.
* On a Simple site, you can enable and disable Carousel from Settings > Media: `/wp-admin/options-media.php`
